### ISSUE TITLE
Chrome container version 3.141.59-neon breaks proxy PAC file support

### ISSUE THREAD

**Issue by cedricroijakkers** [SeleniumHQ/docker-selenium#902]
## 🐛 Bug Report

With docker container version 3.141.59-neon (untested with mercury/lithium/krypton/iron/hafnium) but unable to reproduce with 3.141.59-gold, it seems that Chrome no longer uses the proxy PAC file that can be configured as a desired capability.

## To Reproduce

Set up docker-compose file with one hub and at least one node:

```
version: '3.7'

services:
  selenium-hub:
    container_name: "selenium-hub"
    image: selenium/hub:3.141.59
    environment:
      TZ: "Europe/Amsterdam"
    ports:
      - 4444:4444
    restart: unless-stopped
    logging:
      driver: "json-file"
      options:
        max-size: "50m"
        max-file: "3"
  chrome-grid1:
    container_name: "chrome-grid1"
    image: selenium/node-chrome:3.141.59-neon
    environment:
      HUB_HOST: "fqdn-of-host-machine"
      REMOTE_HOST: "http://fqdn-of-host-machine:5555"
      TZ: "Europe/Amsterdam"
    ports:
      - 5555:5555
    volumes:
      - /dev/shm:/dev/shm
      - /data/pac:/tmp/pac
    restart: unless-stopped
    logging:
      driver: "json-file"
      options:
        max-size: "50m"
        max-file: "3"
```

As you can see, a PAC file is loaded in each container with the following line `/data/pac:/tmp/pac`. This PAC file contains some URLs that need to go over a proxy, because of network limitations. Examples of a PAC file are here: https://findproxyforurl.com/example-pac-file/

Now, create a test with desired capabilities, including one to add a proxy PAC file to the browser (Chrome in this case). This code is a Python example snippet:

```
        self.desired_caps = dict()
        self.desired_caps['browserName'] = browser_name
        self.desired_caps['platformName'] = 'ANY'
        self.desired_caps['platform'] = 'ANY'
        self.desired_caps['version'] = 'ANY'
        self.desired_caps['newCommandTimeout'] = 30
        self.desired_caps['proxy'] = {
            'proxyType': 'PAC',
            'proxyAutoconfigUrl': 'file:///tmp/pac/proxy.pac'
        }
```

And pass these desired capabilities on to the webdriver:

```
self.driver = webdriver.Remote(self.selenium_remote_url, self.desired_caps, options=ext)
```

Kick off the test towards the Selenium grid, and try to reach a URL that is listed in the PAC file to pass over a proxy.

## Expected behavior

The browser will contact the proxy, and will send the request for the URL that matches in the PAC file to the specified proxy

## Actual behavior

It seems like the PAC file is totally ignored in the browser, and since our network has no direct internet connection, this results in a timeout to load the page.

When rolling back the container to version 3.141.59-gold, the issue is resolved.

## Environment

OS: CentOS Linux 7.6.1810 with kernel 3.10.0-957.10.1.el7.x86_64
Docker version: 18.09.5, build e8ff056
Docker-Selenium image version: node-chrome:3.141.59-neon (image id 0fde6f9b6c69)
Docker-compose version: 1.23.2, build 1110ad01

**Comment by diemol** [comment]
@cedricroijakkers, well on the Selenium side there are no differences between https://github.com/SeleniumHQ/docker-selenium/releases/tag/3.141.59-neon and https://github.com/SeleniumHQ/docker-selenium/releases/tag/3.141.59-gold. The difference is that the `gold` tag has Chrome 72.0.3626.81 and ChromeDriver 2.46, whereas the `neon` tag has Chrome 74.0.3729.108 and ChromeDriver 74.0.3729.6.

Have you tried setting it up only with the Selenium jar and Chrome + ChromeDriver? Or just with Chrome and ChromeDriver? I highly doubt that it is something related to Selenium or the docker containers.

**Comment by cedricroijakkers** [comment]
Raised issue with the ChromeDriver team: https://bugs.chromium.org/p/chromedriver/issues/detail?id=2882
Since nothing changed on Selenium side, the most likely candidate is ChromeDriver now.

**Comment by cgoldberg** [comment]
closing since it's tracked as chromedriver issue.

**Comment by cedricroijakkers** [comment]
I have managed to track down the issue and work around it. The root cause is this change in Chromium:

https://bugs.chromium.org/p/chromium/issues/detail?id=839566&q=pac&colspec=ID%20Pri%20M%20Stars%20ReleaseBlock%20Component%20Status%20Owner%20Summary%20OS%20Modified

Chrome no longer allows proxy PAC files to be configured in as a file:// URL, but it really needs to be served over http or https now. There is a workaround to disable the NetworkService (--disable-features=NetworkService), but this is bound to go away soon as well.

So I have worked around it by setting up a small http server to serve the pac file, and now my tests are running again.

Issue can be closed, I've added this info here for future people running into the same issue.
